src/Kernel32.Desktop/Kernel32.cs

```diff
@@ -628,10 +892,11 @@ public static partial class Kernel32
         /// </returns>
         /// <remarks>Minimum OS: Windows Vista / Windows Server 2008.</remarks>
         [DllImport(api_ms_win_core_psapi_l1_1_0, SetLastError = true, CharSet = CharSet.Unicode)]
-        public static extern bool QueryFullProcessImageName(
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern unsafe bool QueryFullProcessImageName(
             SafeObjectHandle hProcess,
             QueryFullProcessImageNameFlags dwFlags,
-            StringBuilder lpExeName,
+            [Friendly(FriendlyFlags.Out | FriendlyFlags.Array)] char* lpExeName,
```

Comment:
I did worry about the broken return but just used the opportunity to raise the `StringBuilder` issue. I think having both is a nice alternative (but with proper return handling, I am not proponent of back-porting bugs). Then I can mark it as something to remove on mark-2 as I asked before.